SCClassLibrary/SCDoc/SCDocRenderer.sc

```diff
@@ -823,14 +827,23 @@ SCDocHTMLRenderer {
 				};
 			},
 			\SUBSECTION, {
-				stream << "<h3><a class='anchor' name='" << this.escapeSpacesInAnchor(node.text)
-				<< "'>" << this.escapeSpecialChars(node.text) << "</a></h3>\n";
+				if(this.escapeSpacesInAnchor(node.text)[0..4] == ":sub:") {
```

Comment:
It seems to me that you are putting part of what should be in the analysis into the generator.

Wouldn't it be better if there were a token `\SUBSUBSECTION` that is generated for `:sub: ` ?

Reply:
Yes, a correct code should be different. 

I have tried several times to get it right but have failed:
https://github.com/supercollider/supercollider/issues/6131

The problem is that the \SUBSUBSECTION token already exists, but is associated with the `method::` tag. 
I also tried the hack with AI, but the code is too big to do it....

Not knowing C++, I did it the wrong way.

Those who know C++ do not need this feature, those who do not know C++ need this feature. The beginning of a tragedy. If someone gives me a hint, I want to try it again the right way.

Reply:
https://github.com/supercollider/supercollider/blob/fb73f7190e741128a6b1c016020cdffc939c32d0/SCDoc/SCDoc.y#L188-L210

According to Gemini 1.5 pro, these parts should be changed to the following to allow the use of the `subsubsection` tag:

```
subsubsection: SUBSUBSECTION words2 eol optsubsubsections { $$ = doc_node_make_take_children("SUBSUBSECTION", $2, $4); }
             | METHOD methnames optMETHODARGS eol methodbody
               {
                 $2->id = "METHODNAMES";
                 $$ = doc_node_make(method_type,$3,$2);
                 doc_node_add_child($$, $5);
               }
             | COPYMETHOD words eol {
                 if ( !(strchr($2, ' ')) ) {
                   yyerror("COPYMETHOD requires 2 arguments (class name and method name)");
                   YYERROR;
                 }

                 $$ = doc_node_make(
                 stringEqual(method_type, "CMETHOD") ? "CCOPYMETHOD"
                                                     : (stringEqual(method_type, "IMETHOD") ? "ICOPYMETHOD"
                                                                                                : "COPYMETHOD"),
                 $2, NULL
                 ); }
             | PRIVATE commalist eoleof { $$ = doc_node_make_take_children( stringEqual(method_type, "CMETHOD") ? "CPRIVATE"
                                                                                                                     : "IPRIVATE",
                 NULL, $2); }
;
```
However, I am unable to test it because the SC I built does not work properly. If I can find a way to build a bug-free SC under MacOS 14.4.1, I will try it.

Reply:
I have tried again to modify C++ code as follows:

1.  changing the following code block
   https://github.com/supercollider/supercollider/blob/fb73f7190e741128a6b1c016020cdffc939c32d0/SCDoc/SCDoc.y#L188-L210
   with the help of Gemini 1.5 pro as follows:  
   ```
   subsubsection: SUBSUBSECTION words2 eol optsubsubsections { $$ =    doc_node_make_take_children("SUBSUBSECTION", $2, $4); }
             | METHOD methnames optMETHODARGS eol methodbody
               {
                 $2->id = "METHODNAMES";
                 $$ = doc_node_make(method_type,$3,$2);
                 doc_node_add_child($$, $5);
               }
             | COPYMETHOD words eol {
                 if ( !(strchr($2, ' ')) ) {
                   yyerror("COPYMETHOD requires 2 arguments (class name and method name)");
                   YYERROR;
                 }

                 $$ = doc_node_make(
                 stringEqual(method_type, "CMETHOD") ? "CCOPYMETHOD"
                                                     : (stringEqual(method_type, "IMETHOD") ? "ICOPYMETHOD"
                                                                                                : "COPYMETHOD"),
                 $2, NULL
                 ); }
             | PRIVATE commalist eoleof { $$ = doc_node_make_take_children( stringEqual(method_type, "CMETHOD") ? "CPRIVATE"
                                                                                                                     : "IPRIVATE",
                 NULL, $2); }
;
```
2. adding the following code 
   ```
   (?i:[ \t]*subsubsection::[ \t]*)    return SUBSUBSECTION;
   ```
   after https://github.com/supercollider/supercollider/blob/8dc8241d21d37c40424f32ddec3f879127aa6c8c/SCDoc/SCDoc.l#L71
   and then
3. adding the following code
   ```
   <metadata,eat>(?i:[ \t]*subsubsection::[ \t]*)    BEGIN(metadata); return SUBSUBSECTION;
   ```
   after https://github.com/supercollider/supercollider/blob/8dc8241d21d37c40424f32ddec3f879127aa6c8c/SCDoc/SCDoc.l#L149

However, the following error still appears when using `subsubsection::`:
> syntax error, unexpected ::, expecting end of file

Reply:
The original commit has been removed.
This PR is now implemented correctly after modifying the following two files
- `SCDoc/SCDoc.l`
- `SCDoc/SCDoc.y`.

I have not written a help document yet. I do not think a tutorial is needed for this implementation.
The `SCDocSyntax.schelp` can be updated if this change is approved.